agent/src/main/java/com/teamscale/jacoco/agent/options/ClasspathUtils.java

```diff
@@ -0,0 +1,38 @@
+package com.teamscale.jacoco.agent.options;
+
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** Handles parsing a .txt file with classpath pattern separated by newlines. */
+public class ClasspathUtils {
+
+	/** Replaces all txt files in the given list with the file names written in the txt file separated by new lines. */
+	public static List<File> resolveClasspathTextFiles(String key, FilePatternResolver filePatternResolver,
+													   List<String> classDirectoriesOrZips) throws AgentOptionParseException {
```

Comment:
rename `classDirectoriesOrZips` to `patterns`? It's could also contain txt files now :-)